core/src/main/java/com/linecorp/armeria/client/DnsResolverGroupBuilder.java

```diff
@@ -75,14 +85,119 @@ public DnsResolverGroupBuilder resolvedAddressTypes(
         return this;
     }
 
+    /**
+     * Sets whether to enable auto refresh for expired {@link DnsRecord}s.
+     * This option is enabled by default.
+     *
+     * <p>If this option is disabled, the expired {@link DnsRecord} is removed from the {@link DnsCache} and a
+     * new {@link DnsQuery} will be executed when the next request to the domain is made.
+     */
+    @UnstableApi
+    public DnsResolverGroupBuilder enableAutoRefresh(boolean autoRefresh) {
```

Comment:
Question) Is it possible to just remove this option?

I think it could be awkward if `autoRefresh = true`, but users still disable by `autoRefreshTimeout == 0`.

I'm wondering if we can instead express this by...
- autoRefreshTimeout == Long.MAX_VALUE: always refresh
- autoRefreshTimeout <= 0 or negative number: don't refresh

I guess if we also set `autoRefreshBackoff = Backoff.ofDefault()` by default, then we don't have to worry about nullability either.

Reply:
It is possible but I don't think it is efficient. 
Although this option is removed, we need to still evaluate the refresh timeout function because it should be called whenever DNS records expire.

If autoRefresh is explicitly disabled, we can skip the process.

Reply:
> I guess if we also set autoRefreshBackoff = Backoff.ofDefault() by default, then we don't have to worry about nullability either.

Originally, it was not null but it became nullable by the following comment.
https://github.com/line/armeria/pull/4400#discussion_r962903309

Reply:
Couldn't we compare functions by reference if the concern is efficiency?

e.g.
```
const DISABLED = () -> LONG.MAX_VALUE
...

if (autoRefreshTimeout == DISABLED) ...
```

Reply:
It seems possible if we change the contract of `autoRefreshTimeout(Duration)` and compare the value to set the pre-defined function to compare.

But the API could be less future-proof when we want to change the default value of `autoRefresh` or the default timeout.
For example, `enableAutoRefresh(true)` can follow our default timeout but `autoRefreshTimeout(Long.MAX_VALUE)` couldn't.

Reply:
I see, I guess we could also maybe define a `DEFAULT_TIMEOUT` function, but I agree this can become unnecessarily complex easily.

My original thought was users may want to switch between `autoRefresh=true` and `autoRefresh=false` dynamically, but I guess it's unclear whether this will be useful.